### PR TITLE
Remove BuiltInFunctionOnMetaEnableEpoch and FixWaitingList

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Obsolete/unused flags : `BuiltInFunctionOnMetaEnableEpoch`, `WaitingListFixEnableEpoch`
- There was a bug in  `sharding/nodesCoordinator/indexHashedNodesCoordinator.go` when computing `computeNodesConfigFromList` for previous nodes config. When having a **leaving** node, `addValidatorToPreviousMap` would always consider that node was **eligible**.
- Linter errors
  
## Proposed changes
- Removed `BuiltInFunctionOnMetaEnableEpoch` + `WaitingListFixEnableEpoch`
- Used implemented `flagWaitingListFix`  in `addValidatorToPreviousMap`, but from `flagStakingV4Started` and completely removed it from `hashValidatorShuffler`. This is mandatory, since an auction unstaked node, would've been considered as previously elgibile otherwise. Therefore, in case of `forced`/`actually remaining` nodes, that auction node could've been directly sent to eligible nodes, without being selected with sufficient topUp

## Testing procedure
- Integration tests

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 1]
```diff
@@ -753,7 +753,7 @@ func (ihnc *indexHashedNodesCoordinator) computeNodesConfigFromList(
 	newNodesList := make([]Validator, 0)
 	auctionList := make([]Validator, 0)
 
-	if ihnc.flagWaitingListFix.IsSet() && previousEpochConfig == nil {
+	if previousEpochConfig == nil {
```
Comment (SebastianMarian) [inline sharding/nodesCoordinator/indexHashedNodesCoordinator.go:756]: if ihnc.flagStakingV4Started.IsSet() && previousEpochConfig == nil {   ?

File: common/enablers/epochFlags.go [hunk 2]
```diff
@@ -613,10 +599,10 @@ func (holder *epochFlagsHolder) IsCheckTransferFlagEnabled() bool {
 	return holder.optimizeNFTStoreFlag.IsSet()
 }
 
-// IsTransferToMetaFlagEnabled returns true if builtInFunctionOnMetaFlag is enabled
-// this is a duplicate for BuiltInFunctionOnMetaEnableEpoch needed for consistency into vm-common
+// IsTransferToMetaFlagEnabled returns false
+// This is used for consistency into vm-common
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply common/enablers/epochFlags.go:603]: Will add a task an delete it on vm-common as well then, but on a separate PR

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go (hunk 1 above)
Comment (mariusmihaic) [reply sharding/nodesCoordinator/indexHashedNodesCoordinator.go:756]: After discussions with @AdoAdoAdo , we will delete this "previousEpochConfig" completely in the next PR anyway, but I'll add it for consistency

**Review by sasurobert — APPROVED** [review]

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go (hunk 1 above)
Comment (SebastianMarian) [reply sharding/nodesCoordinator/indexHashedNodesCoordinator.go]: Ok

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 3]
```diff
@@ -829,11 +832,33 @@ func (ihnc *indexHashedNodesCoordinator) computeNodesConfigFromList(
...
-	eligibleMap[currentValidatorShardId] = append(eligibleMap[currentValidatorShardId], currentValidator)
+	if !ihnc.flagStakingV4Started.IsSet() {
+		eligibleMap[currentValidatorShardId] = append(eligibleMap[currentValidatorShardId], currentValidator)
+		return
+	}
+
+	found, shardId := searchInMap(previousEpochConfig.eligibleMap, currentValidator.PubKey())
+	if found {
+		log.Debug("leaving node found in", "list", "eligible", "shardId", shardId)
+		eligibleMap[shardId] = append(eligibleMap[currentValidatorShardId], currentValidator)
```
Comment (SebastianMarian) [inline sharding/nodesCoordinator/indexHashedNodesCoordinator.go:849]: ...append(eligibleMap[shardId] ?

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 4]
```diff
@@ -829,11 +832,33 @@ func (ihnc *indexHashedNodesCoordinator) computeNodesConfigFromList(
...
+		return
+	}
+
+	found, shardId = searchInMap(previousEpochConfig.waitingMap, currentValidator.PubKey())
+	if found {
+		log.Debug("leaving node found in", "list", "waiting", "shardId", shardId)
+		waitingMap[shardId] = append(waitingMap[currentValidatorShardId], currentValidator)
```
Comment (SebastianMarian) [inline sharding/nodesCoordinator/indexHashedNodesCoordinator.go:856]: ...append(waitingMap[shardId] ?

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 5]
```diff
@@ -829,11 +832,33 @@ func (ihnc *indexHashedNodesCoordinator) computeNodesConfigFromList(
...
+		return
+	}
+
+	log.Debug("leaving node not in eligible or waiting, probably was in auction/inactive/jailed",
```
Comment (SebastianMarian) [inline sharding/nodesCoordinator/indexHashedNodesCoordinator.go:860]: currentValidatorShardId instead shardId. I think in this point shardId will have not value set, probably 0, as found were false in both searching above.

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**